doc/developer/core_developer.rst

```diff
@@ -0,0 +1,167 @@
+often need more handholding, having little to no git
+experience. Repeat yourself liberally, and, if you don’t recognize a
+contributor, point them to our development guide, or other GitHub
+workflow tutorials around the web. Do not assume that they know how
+GitHub works (e.g., many don't realize that adding a commit
+automatically updates a pull request). Gentle, polite, kind
+encouragement can make the difference between a new core developer and
+an abandoned pull request.
+
+When reviewing, focus on the following:
+
+1. **API:** The API is what users see when they first use
+   NetworkX. APIs are difficult to change once released, so
+   should be simple, `functional
+   <https://en.wikipedia.org/wiki/Functional_programming>`__ (i.e. not
+   carry state), consistent with other parts of the library, and
+   should avoid modifying input variables.  Please familiarize
+   yourself with the project's :ref:`deprecation_policy`.
+
+2. **Documentation:** Any new feature should have a gallery
+   example, that not only illustrates but explains it.
+
+3. **The algorithm:** You should understand the code being modified or
+   added before approving it.  (See `Merge Only Changes You
+   Understand`_ below.) Implementations should do what they claim,
+   and be simple, readable, and efficient.
+
+4. **Tests:** All contributions to the library *must* be tested, and
+   each added line of code should be covered by at least one test. Good
+   tests not only execute the code, but explores corner cases.  It is tempting
+   not to review tests, but please do so.
+
+Other changes may be *nitpicky*: spelling mistakes, formatting,
+etc. Do not ask contributors to make these changes, and instead
+make the changes by `pushing to their branch
+<https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/committing-changes-to-a-pull-request-branch-created-from-a-fork>`__,
+or using GitHub’s `suggestion
+<https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/commenting-on-a-pull-request>`__
+`feature
+<https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/incorporating-feedback-in-your-pull-request>`__.
+(The latter is preferred because it gives the contributor a choice in
+whether to accept the changes.)
+
+Our default merge policy is to squash all PR commits into a single
+commit. Users who wish to bring the latest changes from ``master``
+into their branch should be advised to merge, not to rebase.  Even
+when merge conflicts arise, don’t ask for a rebase unless you know
+that a contributor is experienced with git. Instead, rebase the branch
+yourself, force-push to their branch, and advise the contributor on
+how to force-pull.  If the contributor is no longer active, you may
+take over their branch by submitting a new pull request and closing
+the original. In doing so, ensure you communicate that you are not
+throwing the contributor's work away!
```

Comment:
You could perhaps add something about the [`Co-authored-by:`](https://docs.github.com/en/github/committing-changes-to-your-project/creating-a-commit-with-multiple-authors) keyword for commit messages to highlight to reviewers how they could make the attribution more explicit.